### PR TITLE
Change default output array type for `JLD2OutputWriter` to `Array{Float64}`

### PR DESCRIPTION
**Pull request by glwagner**
While `Array{Float32}` is neat because user output is smaller, it's a bad default because it often leads to confusing results, like the fact that outputing `Field(Integral(c))` is different (possibly by `eps(Float32)`) than computing `Integral(c_output)` in post processing when `Field(Integral(c))` is close to 0.

Thus @xkykai and I propose to make the default `Array{Float64}`. We could regard outputting with `Array{Float32}` as an "optimization" which is premature as a default.

### CODE REVIEW

**Comment by glwagner** [comment]
cc @navidcy we also ran into this problem with ParameterEstimocean once

**Comment by navidcy** [comment]
> cc @navidcy we also ran into this problem with ParameterEstimocean once

I remember… took days to figure out what was happening

**Review by navidcy — APPROVED** [review]

**Comment by navidcy** [comment]
Only for JLD2OutputeWriter?

**Comment by glwagner** [comment]
> Only for JLD2OutputeWriter?

Good point.

**Review by navidcy — APPROVED** [review]

File: test/test_netcdf_output_writer.jl [hunk 1]
```diff
@@ -196,12 +196,12 @@ function test_thermal_bubble_netcdf_output(arch)
...
-    @test eltype(ds2["yC"]) == Float32
-    @test eltype(ds2["yF"]) == Float32
-    @test eltype(ds2["zC"]) == Float32
-    @test eltype(ds2["zF"]) == Float32
+    @test eltype(ds2["xC"]) == Float64
+    @test eltype(ds2["xF"]) == Float64
+    @test eltype(ds2["yC"]) == Float64
+    @test eltype(ds2["yF"]) == Float64
+    @test eltype(ds2["zC"]) == Float64
+    @test eltype(ds2["zF"]) == Float64
```
Comment (navidcy) [inline test/test_netcdf_output_writer.jl:204]: `ds2` is `nc_sliced_writer` which was constructed with explicit `array_type = Array{Float32}`

Comment (navidcy) [inline test/test_netcdf_output_writer.jl:204]:
```suggestion
    @test eltype(ds2["xC"]) == Float32
    @test eltype(ds2["xF"]) == Float32
    @test eltype(ds2["yC"]) == Float32
    @test eltype(ds2["yF"]) == Float32
    @test eltype(ds2["zC"]) == Float32
    @test eltype(ds2["zF"]) == Float32
```

**Comment by navidcy** [comment]
(fyi I stopped the tests because everything was clogged... all tests in PRs have been running for 6-7 hrs... so perhaps restart when there is less load)

**Comment by navidcy** [comment]
merge?
we didn't bump any patch release, is this OK?

**Comment by glwagner** [comment]
Let's bump

**Comment by navidcy** [comment]
why 0.79.3? we are only up to 0.79.1... am I missing something?

**Comment by glwagner** [comment]
> why 0.79.3? we are only up to 0.79.1... am I missing something?

you;re right

**Pull request merged**